server/chat-commands/core.ts

```diff
@@ -726,6 +729,23 @@ export const commands: ChatCommands = {
 		});
 	},
 
+	'!updatesettings': true,
+	updatesettings(target, room, user) {
+		const settings: Partial<UserSettings> = {};
+		try {
+			const raw = JSON.parse(target);
+			if (typeof raw !== 'object' || Array.isArray(raw) || !raw) {
+				this.errorReply("/updatesettings expects JSON encoded object.");
+			}
+			for (const setting of User.SETTINGS) {
+				if (setting in raw) settings[setting] = settings !== 'blockPMs' ? !!raw[setting] : raw[setting];
```

Comment:
also while i'm nitpicking, i think it's possible to cause a crash in `authAtLeast` if you pass something like `{blockPMs: {toString: null}}`

Reply:
also i think this should be `setting !== 'blockPMs'`? i'm not sure why typescript isn't complaining about that.

Reply:
"nitpicking" aka fixing my broken code! thanks. And yeah, i was worried about blockPMs because it takes a string here. Possibly need to include a set with all the possible values it can take (and give blockPMs a stricter type than `boolean | string` while we're at it?)

Reply:
OK, I've actually typed `blockPMs` properly (I think?) and filter it correctly now!